homeassistant/components/tankerkoenig/__init__.py

```diff
@@ -0,0 +1,204 @@
+    CONF_SCAN_INTERVAL,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import CONF_FUEL_TYPES, CONF_STATIONS, DOMAIN, FUEL_TYPES, NAME
+from .sensor import FuelPriceSensor
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_RADIUS = 2
+SCAN_INTERVAL = timedelta(minutes=30)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_API_KEY): cv.string,
+                vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
+                vol.Optional(CONF_FUEL_TYPES, default=FUEL_TYPES): vol.All(
+                    cv.ensure_list, [vol.In(FUEL_TYPES)]
+                ),
+                vol.Inclusive(
+                    CONF_LATITUDE,
+                    "coordinates",
+                    "Latitude and longitude must exist together",
+                ): cv.latitude,
+                vol.Inclusive(
+                    CONF_LONGITUDE,
+                    "coordinates",
+                    "Latitude and longitude must exist together",
+                ): cv.longitude,
+                vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS): vol.All(
+                    cv.positive_int, vol.Range(min=1)
+                ),
+                vol.Optional(CONF_STATIONS, default=[]): vol.All(
+                    cv.ensure_list, [cv.string]
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set the tankerkoenig component up."""
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+
+    _LOGGER.debug("Setting up platform")
+
+    tankerkoenig = await hass.async_add_executor_job(
+        partial(TankerkoenigData, hass, conf)
+    )
+
+    if not tankerkoenig.entity_list:
+        _LOGGER.error("Could not find any fuel station to track")
+        return False
+
+    hass.data[DOMAIN] = tankerkoenig
+    async_track_time_interval(hass, tankerkoenig.update, conf[CONF_SCAN_INTERVAL])
+
+    hass.async_create_task(
+        async_load_platform(
+            hass, SENSOR_DOMAIN, DOMAIN, discovered=None, hass_config=conf
+        )
+    )
+
+    return True
+
+
+class TankerkoenigData:
+    """Get the latest data from the API."""
+
+    def __init__(self, hass, conf):
+        """Initialize the data object."""
+        self._api_key = conf[CONF_API_KEY]
+        self._entities = {}
+        self._fuel_types = conf[CONF_FUEL_TYPES]
+
+        latitude = conf.get(CONF_LATITUDE, hass.config.latitude)
+        longitude = conf.get(CONF_LONGITUDE, hass.config.longitude)
+        radius = conf[CONF_RADIUS]
+
+        additional_stations = conf[CONF_STATIONS]
+
+        _LOGGER.debug("Fetching data for (%s,%s) rad: %s", latitude, longitude, radius)
+
+        try:
+            data = pytankerkoenig.getNearbyStations(
```

Comment:
We don't want side effects in init method. Please either pass this instance to `TankerkoenigData` or move this to a dedicated setup method.